### PR TITLE
Port reverse(varbinary) from Prestosql to Prestodb

### PR DESCRIPTION
**Pull request by EmmyEmmy**
Port commit from Prestosql https://github.com/prestosql/presto/commit/ac585502abd05b413644bb65a30a728b0213f19a to Prestodb

### CODE REVIEW

**Review by EmmyEmmy — COMMENTED** [review]
Port reverse(varbinary) from Prestosql

**Review by kaikalur — CHANGES_REQUESTED** [review]
Just add a test for null

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java [hunk 1]
```diff
@@ -511,4 +511,13 @@ private static SqlVarbinary varbinary(int... bytesAsInts)
         }
         return new SqlVarbinary(bytes);
     }
+
+    @Test
+    public void testReverse()
```
Comment (kaikalur) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java:516]: Also add a couple of tests that check: 

reverse(reverse(x)) = x

**Review by kaikalur — CHANGES_REQUESTED** [review]
One more test

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java (hunk 1 above)
Comment (EmmyEmmy) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java:516]: Added

**Review by EmmyEmmy — COMMENTED** [review]
Added two more tests `reverse(reverse())` and `null`

**Comment by rongrong** [comment]
> Just add a test for null

Test null is not necessary because functions are by default treated as "return null on null" in presto so the user logic will not be triggered if input is null.

**Comment by rongrong** [comment]
@EmmyEmmy please squash all changes into a single commit and sign the CLA. Thanks!

**Comment by kaikalur** [comment]
> > Just add a test for null
> 
> Test null is not necessary because functions are by default treated as "return null on null" in presto so the user logic will not be triggered if input is null.

Good to test still.

**Comment by rongrong** [comment]
> > > Just add a test for null
> > 
> > 
> > Test null is not necessary because functions are by default treated as "return null on null" in presto so the user logic will not be triggered if input is null.
> 
> Good to test still like a harness.

We do test that. The logic has nothing to do with logic introduced in this PR. I'm not against the test case. Just pointing out that it's not necessary.

**Comment by rongrong** [comment]
Please also change the commit message to `Add reverse function for varbinary` and add `Co-Authored-By` of the original author to commit message. You can find examples of this in `git log`. Thanks!

**Review by EmmyEmmy — COMMENTED** [review]
Squashed to 1 commit

**Review by kaikalur — APPROVED** [review]
LGTM

**Review by rongrong — APPROVED** [review]

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java [hunk 2]
```diff
@@ -511,4 +511,15 @@ private static SqlVarbinary varbinary(int... bytesAsInts)
...
+    {
+        assertFunction("REVERSE(CAST(null AS VARBINARY))", VARBINARY, sqlVarbinary(""));
```
Comment (rongrong) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java:518]: this one should probably be `assertFunction("REVERSE(CAST(null AS VARBINARY))", VARBINARY, null);` or something like that?

Comment (EmmyEmmy) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java]: Yup, thanks =) Sent another iter

**Pull request merged**